pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,250 @@
+package browse
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type browser interface {
+	Browse(string) error
+}
+
+type BrowseOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    browser
+
+	SelectorArg   string
+	AdditionalArg string
+
+	ProjectsFlag bool
+	WikiFlag     bool
+	SettingsFlag bool
+	BranchFlag   bool
+	//LineFlag     bool
+}
+
+type exitCode int
+
+const (
+	exitSuccess      exitCode = 0
```

Comment:
in `gh`, a command doesn't need to worry about exit codes. It returns an error and, if the code in `cmd/gh/main.go` cares, it can select the appropriate exit code based on the type of the error it sees.

Instead of defining `exitCode` like this, you can just have your functions return an `error`, which is a built in Go type.

To work with errors, look around the codebase for examples of either `fmt.ErrorfF` or `errors.New`.

Reply:
Good to note, I'm glad we have a pr posted now to so we can start resolving issues such as this. We'll add these as a priority on the product backlog to help round out the feature. Thanks for all the feedback you gave, we'll try to update all of these asap!